src/mlpack/tests/q_learning_test.cpp

```diff
@@ -251,4 +251,69 @@ BOOST_AUTO_TEST_CASE(AcrobotWithDQN)
   BOOST_REQUIRE_EQUAL(success, true);
 }
 
+//! Test DQN in Mountain Car task.
+BOOST_AUTO_TEST_CASE(MountainCarWithDQN)
+{ 
+  // Set up the network.
+  FFN<MeanSquaredError<>, GaussianInitialization> model(MeanSquaredError<>(),
+      GaussianInitialization(0, 0.001));
+  model.Add<Linear<>>(2, 64);
+  model.Add<ReLULayer<>>();
+  model.Add<Linear<>>(64, 32);
+  model.Add<ReLULayer<>>();
+  model.Add<Linear<>>(32, 3);
+
+  // Set up the policy and replay method.
+  GreedyPolicy<MountainCar> policy(1.0, 1000, 0.1);
+  RandomReplay<MountainCar> replayMethod(20, 10000);
+
+  TrainingConfig config;
+  config.StepSize() = 0.0001;
+  config.Discount() = 0.9;
+  config.TargetNetworkSyncInterval() = 100;
+  config.ExplorationSteps() = 100;
+  config.DoubleQLearning() = false;
+  config.StepLimit() = 400;
+
+  // Set up DQN agent.
+  QLearning<MountainCar, decltype(model), AdamUpdate, decltype(policy)>
+      agent(std::move(config), std::move(model), std::move(policy),
+      std::move(replayMethod));
+
+  arma::running_stat<double> averageReturn;
+  size_t episodes = 0;
+  bool converged = true;
+  while (true)
+  {
+    double episodeReturn = agent.Episode();
+    averageReturn(episodeReturn);
+    episodes += 1;
+
+    if (episodes > 1000)
+    {
+      Log::Debug << "Mountain Car with DQN failed." << std::endl;
+      converged = false;
+      break;
+    }
+
+    /**
+     * Set a threshold of -370 to check a convergence.
```

Comment:
I think it is simply `check convergence`.

Reply:
I meant we should just remove the 'a' :)

Reply:
Haha!! Thanks. :)